odoo/service/model.py

```diff
@@ -165,7 +166,10 @@ def execute_cr(cr, uid, obj, method, *args, **kw):
     recs = odoo.api.Environment(cr, uid, {}).get(obj)
     if recs is None:
         raise UserError(_("Object %s doesn't exist") % obj)
-    return odoo.api.call_kw(recs, method, args, kw)
+    result = odoo.api.call_kw(recs, method, args, kw)
+    for l in iterate(result, lazy):
+        _0 = l._value # evaluate and cache while we still have a valid cursor
```

Comment:
I would have put a comment above the `for` loop: "force evaluation of lazy values before closing the cursor".
I was wondering whether `l._value` might still contain lazy values, but that's not the case with `lazy_name_get`.
So that's fine.